commander/src/utils/reader.ts

```diff
@@ -232,107 +189,120 @@ const castValue = (
 	return val;
 };
 
-export const transformAsset = (
-	schema: Schema,
-	data: Record<string, string>,
-): Record<string, unknown> => {
-	const propertySchema = Object.values(schema.properties);
-	const assetData = {} as Record<string, unknown>;
-	return Object.entries(data).reduce((acc, curr, index) => {
-		const propSchema = propertySchema[index] as { type: string; dataType: string };
-		// Property schema type can be scalar(string, bool, etc..) or structural(object, array)
-		const schemaType = propSchema.type || propSchema.dataType;
-		acc[curr[0]] = castValue(curr[1], schemaType);
-		return acc;
-	}, assetData);
-};
+const castArray = (items: string[], schemaType: string): string[] | number[] | bigint[] => {
+	if (schemaType === 'uint64' || schemaType === 'sint64') {
+		return items.map(i => BigInt(i));
+	}
 
-export const transformNestedAsset = (
-	schema: Schema,
-	data: Array<Record<string, string>>,
-): NestedAsset => {
-	const template = getNestedPropertyTemplate(schema);
-	const result = {} as NestedAsset;
-	const items: Array<Record<string, unknown>> = [];
-	for (const assetData of data) {
-		const [[key, val]] = Object.entries(assetData);
-		const templateValues = template[key];
-		const initData = {} as Record<string, unknown>;
-		const valObject = val.split(',').reduce((acc, curr, index) => {
-			acc[templateValues[index]] = Number.isInteger(Number(curr)) ? Number(curr) : curr;
-			return acc;
-		}, initData);
-		items.push(valObject);
-		result[key] = items;
+	if (schemaType === 'uint32' || schemaType === 'sint32') {
+		return items.map(i => Number(i));
 	}
+
+	return items;
+};
+
+const getNestedParametersFromPrompt = async (property: {
+	name: string;
+	items: { properties: Record<string, unknown> };
+}) => {
+	let addMore = false;
+	const nestedArray: Array<Record<string, unknown>> = [];
+
+	do {
+		const nestedProperties = Object.keys(property.items.properties);
```

Comment:
Maybe you could move `nestedProperties` and `nestedPropertiesCsv` vars outside the do-while loop as their values don't change in the loop. This can improve performance as these operations are not repeated in each loop iteration.

Reply:
[Modified here.](https://github.com/LiskHQ/lisk-sdk/pull/9118/commits/6de94923dd0092a73315b78d3a86ba4edf4bf092)